akka-actor/src/main/scala/akka/actor/Actor.scala

```diff
@@ -481,7 +491,12 @@ trait Actor {
    * @param receive current behavior.
    * @param msg current message.
    */
-  protected[akka] def aroundReceive(receive: Actor.Receive, msg: Any): Unit = receive.applyOrElse(msg, unhandled)
+  protected[akka] def aroundReceive(receive: Actor.Receive, msg: Any): Unit = {
+    // optimization: avoid allocation of lambda
+    if (receive.applyOrElse(msg, Actor.notHandledFun) == Actor.NotHandled) {
```

Comment:
asInstanceOf[AnyRef] then? I am pretty sure that cast is optimized away as it is meaningless.